server/balancer.go

```diff
@@ -340,6 +368,49 @@ func (rb *resourceBalancer) Balance(cluster *clusterInfo) (*balanceOperator, err
 	return rb.doFollowerBalance(cluster, stores, region, follower, newPeer)
 }
 
+func (rb *resourceBalancer) doLeaderTransfer(cluster *clusterInfo) (*balanceOperator, error) {
+	stores := cluster.getStores()
+	region, leader, newLeader := rb.selectLeaderTransferRegion(cluster, stores)
+	if region == nil || leader == nil || newLeader == nil {
+		log.Warn("region cannot be found to do leader transfer")
+		return nil, nil
+	}
+
+	// If region peer count is not equal to max peer count, no need to do leader transfer.
+	if len(region.GetPeers()) != int(cluster.getMeta().GetMaxPeerCount()) {
+		log.Warnf("region peer count %d not equals to max peer count %d, no need to do leader transfer",
+			len(region.GetPeers()), cluster.getMeta().GetMaxPeerCount())
+		return nil, nil
+	}
+
+	if !rb.checkScore(cluster, leader, newLeader, true) {
```

Comment:
no need to check leader diff here? E.g, transferring from store 1 -> 2, now 1 has 11 leaders, 2 has 10 leaders, but 1 and 2 has different capacity so checkScore returns true.